pkg/asset/machines/openstack/machines.go

```diff
@@ -37,24 +37,46 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 	if poolPlatform := pool.Platform.Name(); poolPlatform != openstack.Name {
 		return nil, fmt.Errorf("non-OpenStack machine-pool: %q", poolPlatform)
 	}
-	platform := config.Platform.OpenStack
 
-	az := ""
-	provider, err := generateProvider(clusterID, platform, pool.Platform.OpenStack, osImage, az, role, userDataSecret)
+	mpool := pool.Platform.OpenStack
+	platform := config.Platform.OpenStack
+	trunkSupport, err := checkNetworkExtensionAvailability(platform.Cloud, "trunk")
 	if err != nil {
 		return nil, err
 	}
 
-	if role == "master" {
-		provider.ServerGroupName = clusterID + "-master"
-	}
-
 	total := int64(1)
```

Comment:
it always creates only one spec, we need to change this to `total := int64(len(mpool.Zones))`